examples/advanced/federated-statistics/df_stats/job_api/df_statistics.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Optional
+
+import pandas as pd
+
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_opt.statistics.df.df_core_statistics import DFStatisticsCore
+
+
+class DFStatistics(DFStatisticsCore):
+    def __init__(self, data_path):
+        super().__init__()
+        self.data_root_dir = "/tmp/nvflare/df_stats/data"
+        self.data_path = data_path
+        self.data: Optional[Dict[str, pd.DataFrame]] = None
+        self.data_features = [
+            "Age",
+            "Workclass",
+            "fnlwgt",
+            "Education",
+            "Education-Num",
+            "Marital Status",
+            "Occupation",
+            "Relationship",
+            "Race",
+            "Sex",
+            "Capital Gain",
+            "Capital Loss",
+            "Hours per week",
+            "Country",
+            "Target",
+        ]
+
+        # the original dataset has no header,
+        # we will use the adult.train dataset for site-1, the adult.test dataset for site-2
+        # the adult.test dataset has incorrect formatted row at 1st line, we will skip it.
+        self.skip_rows = {
+            "site-1": [],
+            "site-2": [0],
```

Comment:
Should avoid the hardcode client name "site-1" and "site-2" here. client names can be passed in from the constructor.